app/src/main/java/org/schabi/newpipe/player/playqueue/PlayQueue.java

```diff
@@ -526,7 +526,16 @@ public boolean equals(@Nullable final Object obj) {
             return false;
         }
         final PlayQueue other = (PlayQueue) obj;
-        return streams.equals(other.streams);
+        if (size() != other.size()) {
+            return false;
+        }
+        for (int i = 0; i < size(); i++) {
+            // Check is based on URL
+            if (!streams.get(i).getUrl().equals(other.streams.get(i).getUrl())) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
> Also I noticed that for some reason equals check is not working correctly currently. So I reimplemented it based on urls compare of inner streams (see the first commit).

Why this change? Is it essential to do it in this PR? What do you mean by "not working" and in which cases is this working better?
I guess it is correct to compare play queues based just on stream urls. But [a similar change](https://github.com/TeamNewPipe/NewPipe/commit/2a2c82e73bb2380c07ab848b185952ab183a5502#diff-6dcb0e241835a3f32ba00835e94c64b48b6faf84c0252769617507c1c2fd8371) on `PlayQueueItem`s caused [buffering problems](https://github.com/TeamNewPipe/NewPipe/issues/4940) on duplicate items in queue, finally fixed by #6712. That's why I'd prefer to move this to a different PR, where we could better analyze which differences are to be found in the app if the play queue comparation is changed.

Reply:
I should write more miningfull description of commits, just forgot details about this one. 
If I remember correctly wrong equals checks was the reason of duplicated queues in backstack. It means that the same queue was added to backstack and when you click back you'll see the same stream. 
It's probably something that gives problems others didn't find the cause of.

I don't think the fix could give other problems because the idea of equals check is to be sure that queues are the same. In case of queue we care only about the URLs of all streams contained in a queue.

Reply:
@Stypox see https://github.com/TeamNewPipe/NewPipe/pull/7910